lib/network/api/api.go

```diff
@@ -37,6 +41,29 @@ func NewNetworkHandlerAPI(localNode *node.LocalNode, network network.Network, st
 	}
 }
 
+func renderEventStream(args ...interface{}) ([]byte, error) {
```

Comment:
`renderEventStream()` will be big thing in this PR. I have another idea. How about add new interface method, `Render()` in `APIResource` and `ResourceList`, so `EventStream.Render()` and `WriteJSON()` call it. In this design, we don't need to add cases in `renderEventStream`.

Reply:
> renderEventStream() will be big thing in this PR. I have another idea. How about add new interface method, Render() in APIResource and ResourceList, so EventStream.Render() and WriteJSON() call it. In this design, we don't need to add cases in renderEventStream.

As you know,  `observer.Trigger`  triggers  `BlockXXX`. So The `EventStream.RenderFunc` make resources from blocks.  The reason of type switch exists is that. 

If BlockXXX has `Render()` or  Trigging `HALResource`, Maybe `EventStream` don't need `RenderFunc` IHMO.

```
 if err == nil {
        event := "saved"
        event += " " + fmt.Sprintf("address-%s", b.Address)
        observer.BlockAccountObserver.Trigger(event, b)
    }
```

Reply:
@anarcher You are right, I missed that. We can design better next time.

Reply:
Yep!  We can design better Next time! :-]